backend/contracts/ElectionFactory.sol

```diff
@@ -1,28 +1,51 @@
-//SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: UNLICENSED
 
 pragma solidity ^0.8.0;
 
 contract ElectionFactory {
-    struct Election {
+    struct Candidate {
+        uint8 id;
         string name;
     }
 
-    Election[] public elections;
-
-    constructor() {
+    struct Ballot {
+        string name;
+        string description;
+        uint256 createTime;
+        uint256 votingTime;
+        Candidate[] candidates;
     }
 
-    function createElection(string memory _name) external{
-        Election memory newElection = Election(_name);
-        elections.push(newElection);
+    mapping(uint256 => Ballot) public ballots;
+    uint256 public nextBallotId;
+
+    event CandidateAdded(uint256 ballotId, uint8 id, string name);
+    event ContractInitialized(uint256 ballotId, string name, string description, uint256 createTime, uint256 votingTime, Candidate[] candidates);
+
+    function clearCandidates(uint256 _ballotId) internal {
+        delete ballots[_ballotId].candidates; // just a precaution for any random data that might have left in array
     }
 
-    function getAllElectionNames() external view returns (string[] memory) {
-        uint256 length = elections.length;
-        string[] memory names = new string[](length);
-        for (uint256 i = 0; i < length; i++) {
-            names[i] = elections[i].name;
+    function initialize(string memory _name, string memory _description, uint256 _votingTime, string[] memory _candidateNames) public {
+
+        // check for a number of elements in given array (min. 2 candidates, max. - 10)
+        require(_candidateNames.length >= 2 && _candidateNames.length <= 10, "Invalid number of candidates");
+
```

Comment:
Think about the security of this operation. Currently there is no check who exactly is creating a ballot (it could be any random account on the blockchain). Consider adding a `require` check whether it's the contract creator, who is initializing the transaction. You can checkout [this resource](https://medium.com/coinmonks/learn-solidity-lesson-12-addresses-and-only-owner-5022234848fe) to see what I mean

Reply:
probably is done, pls review

Reply:
Alright, but next time you want me to review, just re-request the review